### ISSUE TITLE
Example of a blurry thumbnail image

### ISSUE THREAD

**Issue by goodman-capu** [Marginal/QLVideo#109]
Example of a blurry thumbnail image

https://github.com/Marginal/QLVideo/issues/108

<img width="147" alt="Screenshot 2023-01-24 at 11 19 17 AM" src="https://user-images.githubusercontent.com/8862302/214387791-c3cb5ad0-2fd4-47a0-82ab-4da0248249e1.png">

---
QLVideo: 2.00
macOS: Version 13.2 (Build 22D49)
Hardware: MacBookPro15,1 x86_64 avx2=yes avx512f=no
mdimport: Imported '*file*.mov' of type 'com.apple.quicktime-movie' with plugIn /System/Library/Spotlight/CoreMedia.mdimporter. 42 attributes returned .
```json
{
  "programs": [
  ],
  "streams": [
    {
      "codec_name": "h264",
      "profile": "Main",
      "codec_type": "video",
      "codec_tag_string": "avc1",
      "width": 1704,
      "height": 824,
      "display_aspect_ratio": "213:103",
      "pix_fmt": "yuv420p",
      "disposition": {
        "default": 1,
        "attached_pic": 0,
        "timed_thumbnails": 0
      }
    }
  ],
  "format": {
    "format_name": "mov,mp4,m4a,3gp,3g2,mj2",
    "duration": "10.233333",
    "size": "635623",
    "bit_rate": "496903",
    "probe_score": 100
  }
}
```

**Comment by goodman-capu** [comment]
I can confirm this issue also exists on another M1 Pro MacBook Pro (macOS 13.2).

**Comment by goodman-capu** [comment]
I can confirm this issue also exists on another Intel i5 MacBook Pro (macOS Big Sur 11.1).

**Comment by xxmmllqqqqccqq** [comment]
Same here on intel i5 MacBook Pro with macOS Big Sur 11.7.2

**Comment by vbwx** [comment]
I haven't seen blurry thumbnail images so far but I can confirm that it's taking QLVideo multiple seconds to generate thumbnails for MP4 files in particular (per file), and if there are many files in a folder that QLVideo has to create thumbnails for, the `thumbnailer` process seems to get stuck and is using up 300% of the CPU.

Running macOS 12.6.3 on an Intel Core i7 MacBook Pro.

**Comment by Kagami** [comment]
> if there are many files in a folder that QLVideo has to create thumbnails for, the thumbnailer process sometimes gets stuck and is using up 300% of the CPU

Yeah, it's not clear what is Finder doing under its hood. I even don't have some thumbnails for mp4 files which should be generated natively... Maybe it goes crazy and iterates over all videos files in Spotlight index, maybe gets some errors during it and just stops. Then maybe it will continue again but not clear when and why some valid files are still missing thumbnail. Very opaque kind of process.
